apis/hive/v1/clusterdeployment_types.go

```diff
@@ -675,6 +676,10 @@ type ClusterIngress struct {
 	// should be used for this Ingress
 	// +optional
 	ServingCertificate string `json:"servingCertificate,omitempty"`
+
+	// HttpErrorCodePages allows configuring custom HTTP error pages using the IngressController object
+	// +optional
+	HttpErrorCodePages configv1.ConfigMapNameReference `json:"httpErrorCodePages,omitempty"`
```

Comment:
I think it would be better if this was 
```suggestion
	HttpErrorCodePages *configv1.ConfigMapNameReference `json:"httpErrorCodePages,omitempty"`
```

Otherwise `omitempty` and `+optional` don't do what you expect them to; and it's also possible to break backward compatibility.

Note that if you make this change, you'll need to add nil checks wherever you reference the field.